camply/search/search_going_to_camp.py

```diff
@@ -0,0 +1,212 @@
+"""
+Going To Camp API search utilities
+"""
+import logging
+from datetime import datetime, time, timedelta
+from typing import List, Optional, Set, Tuple, Union
+
+import pandas as pd
+
+from camply.config import YellowstoneConfig
+from camply.containers import AvailableCampsite, CampgroundFacility, SearchWindow
+from camply.providers import GoingToCampProvider, RecreationDotGov
+from camply.providers.going_to_camp.going_to_camp_provider import RECREATION_AREAS
+from camply.search.base_search import BaseCampingSearch, SearchError
+from camply.utils import make_list
+from camply.utils.logging_utils import log_sorted_response
+
+logger = logging.getLogger(__name__)
+
+
+class SearchGoingToCamp(BaseCampingSearch):
+    """
+    Camping Search Object
+    """
+
+    # noinspection PyUnusedLocal
+    def __init__(
+        self,
+        search_window: Union[SearchWindow, List[SearchWindow]],
+        recreation_area: Optional[Union[List[int], int]] = None,
+        campsites: Optional[Union[List[int], int]] = None,
+        weekends_only: bool = False,
```

Comment:
The `BaseCampingSearch` base class should handle all of this heavy lifting.

Here's how it works inside with Recreation.Gov in the final `get_all_campsites` abstractmethod:

https://github.com/juftin/camply/blob/d1fb5d595c02aab65c3ced44a46f13e1dfda9381/camply/search/search_recreationdotgov.py#L247-L257

The idea is that `found_campsites` array is a list of `AvailableCampsite` objects which may or may not fit the search criteria. Every `AvailableCampsite` has a booking nights length of 1 and the `_filter_date_overlap` will take care of filtering to match dates (including weekends) and `_consolidate_campsites` will handle combining consecutive nights together for the same campsite/campground ID when the user specifies that `nights` > 1. 

Yellowstone was a bit different than this since it allowed you to forward the `nights` param directly to the API to only return bookings matching that.